framework/src/main/java/org/checkerframework/common/aliasing/AliasingTransfer.java

```diff
@@ -83,41 +89,80 @@ public TransferResult<CFValue, CFStore> visitAssignment(
      */
     @Override
     protected void processPostconditions(
-            MethodInvocationNode n, CFStore store, ExecutableElement methodElement, Tree tree) {
-        super.processPostconditions(n, store, methodElement, tree);
-        if (TreeUtils.isEnumSuper(n.getTree())) {
+            Node n, CFStore store, ExecutableElement executableElement, ExpressionTree tree) {
+        MethodInvocationNode methodInvocationNode = null;
+        ObjectCreationNode objectCreationNode = null;
+        if (n instanceof MethodInvocationNode) {
+            methodInvocationNode = (MethodInvocationNode) n;
+        } else if (n instanceof ObjectCreationNode) {
+            objectCreationNode = (ObjectCreationNode) n;
+        } else {
+            throw new BugInCF(
+                    "processPostconditions in AliasingTransfer expects "
+                            + "a MethodInvocationNode or ObjectCreationNode argument; received a "
+                            + n.getClass().getSimpleName());
+        }
+        super.processPostconditions(n, store, executableElement, tree);
+        if (methodInvocationNode != null && TreeUtils.isEnumSuper(methodInvocationNode.getTree())) {
             // Skipping the init() method for enums.
             return;
         }
-        List<Node> args = n.getArguments();
-        List<? extends VariableElement> params = methodElement.getParameters();
+        List<Node> args =
+                methodInvocationNode != null
+                        ? methodInvocationNode.getArguments()
+                        : objectCreationNode.getArguments();
+        List<AnnotatedTypeMirror> params = null;
+        AnnotatedExecutableType annotatedType = factory.getAnnotatedType(executableElement);
+        if (methodInvocationNode != null) {
+            params = annotatedType.getParameterTypes();
```

Comment:
For a method invocation, why isn't this calling `methodFromUse`, like it's done for constructor invocation?
This would take the parameters from the method declaration, without adapting it to the particular use here.

Reply:
Yeah, I think it need to call methodFromUse, the reason why I did not use it is the args of method invocation and constructor are different when passing varargs...